pkg/patterns/addon/pkg/status/conditions.go

```diff
@@ -0,0 +1,64 @@
+package status
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	addonsv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+)
+
+const (
+	AbnormalReason = "ManifestsNotReady"
+	NormalReason   = "Normal"
+	ReadyType      = "Ready"
+)
+
+// SetInProgress set the present condition to a single condition with type "Ready" and status "false". This means
+// the current resources is still reconciling. If any deployment manifests are abnormal, their abnormal status condition will
+// be recorded in the `message` field.
+func SetInProgress(commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	setCondition(metav1.ConditionFalse, commonStatus, abnormalTrueConditions)
+}
+
+// SetReady set the present condition to a single condition with type "Ready" and status "true". This means
+// all the deployment manifests are reconciled.
+func SetReady(commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	setCondition(metav1.ConditionTrue, commonStatus, abnormalTrueConditions)
+}
+
+func setCondition(status metav1.ConditionStatus, commonStatus *addonsv1alpha1.CommonStatus, abnormalTrueConditions []status.Condition) {
+	reason, message := humanMessagefromConditions(abnormalTrueConditions)
+	newCondition := metav1.Condition{
+		Status:  status,
+		Type:    ReadyType,
+		Reason:  reason,
+		Message: message,
+	}
+	meta.SetStatusCondition(&commonStatus.Conditions, newCondition)
+}
+
+// humanMessagefromConditions summarize the kstatus abnormal-true conditions to a reason with human-readable message.
+// The "reason" should be "Normal" if no deployment manifests have abnormal conditions, or "ManifestsNotReady"
+// as long as one deployment manifest has an abnormal condition.
+// The "message" contains each abnormal condition's "reason" and "message".
+// e.g.
+//
+//	 conditions:
+//	 - reason: ManifestsNotReady
+//	   message: |-
+//		    apps/v1, Kind=Deployment/argocd/argocd-repo-server:Deployment does not have minimum availability.
+//		    apps/v1, Kind=Deployment/argocd/argocd-server:Deployment does not have minimum availability.
+func humanMessagefromConditions(conditions []status.Condition) (reason, message string) {
+	if len(conditions) == 0 {
+		return NormalReason, "all manifests are reconciled."
+	}
+	abnormalMessage := ""
+	for _, cond := range conditions {
+		if cond.Message != "" {
+			abnormalMessage += fmt.Sprintln(cond.Message)
```

Comment:
I would just do `abnormalMessage += cond.Message + "\n"`

The reason is that if there's a %s in cond.Message, it'll be treated as a formatting directive and things get messy.

The other option is `fmt.Sprintln("%s", cond.Message)`, but I personally think string concatenation is easier to read in that case.